azure_encryption_helper/encryptor.py

```diff
@@ -0,0 +1,47 @@
+from .aescipher import AESCipher
+from .aeskeywrapper import AESKeyWrapper
+
+import os
+
+class Encryptor:
+
+    def __init__(self, key=None, iv_length=16):
+        if key is None:
+            key = Encryptor.generate_aes_key()
+        self.key = key
+        self.iv_length = iv_length
+        self._update_cipher()
+
+    @staticmethod
+    def generate_aes_key(length=32):
+        return os.urandom(length)
+
+    def configure_wrapper(self, vault, credentials, key_name, key_version):
+        self.wrapper = AESKeyWrapper(vault, credentials, key_name, key_version)
+
+    def set_key(self, key, iv_length=None):
```

Comment:
Perhaps we should keep the key immutable?

Reply:
If we want to keep it immutable, constructor has to accept both wrapped & unwrapped keys.. It feels confusing to me